src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -90,7 +91,7 @@
 
     final Symtab syms;
     final JavacMessages messages;
-    final Names names;
+    private Names names;
```

Comment:
where is this change coming from?

Reply:
When building the JDK, I got the following error:

```
jdk/src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java:950: error: variable names might not have been initialized
                    t.name != names.init &&
                              ^
1 error
```

Therefore I revised the modifier of the `Names names`.

Reply:
It's not a bug, it's a dubious feature, the lambda EG discussed that at length, you can ask Brian about it.

Usually replacing
```
t.name != names.init
```
by
```
  t.name != this.names.init
```
is enough.

Reply:
@forax Thank you for your reply. I tested `t.name != this.names.init` with `final Names names;` just now. The error also occurred. Same as `t.name != names.init`.

And I think that `t.name != names.init` and `t.name != this.names.init` are equal at this situation.